contracts/collections/sg721-updatable/src/contract.rs

```diff
@@ -7,19 +7,33 @@ use cosmwasm_std::{Deps, StdResult};
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::{DepsMut, Env, Event, MessageInfo};
 use cw2::set_contract_version;
+use cw721::msg::Cw721MigrateMsg;
+use cw721::{
+    DefaultOptionalCollectionExtension, DefaultOptionalCollectionExtensionMsg,
+    DefaultOptionalNftExtension, DefaultOptionalNftExtensionMsg,
+};
 use semver::Version;
 use sg721::InstantiateMsg;
+#[allow(deprecated)]
 use sg721_base::msg::CollectionInfoResponse;
 
 use crate::msg::{EnableUpdatableResponse, FrozenTokenMetadataResponse};
 use crate::state::ENABLE_UPDATABLE;
 
-use cw721_base::Extension;
 use cw_utils::nonpayable;
 use sg1::checked_fair_burn;
 use sg721_base::ContractError::Unauthorized;
 use sg721_base::Sg721Contract;
-pub type Sg721UpdatableContract<'a> = Sg721Contract<'a, Extension>;
+pub type Sg721UpdatableContract<'a> = Sg721Contract<
+    'a,
+    DefaultOptionalNftExtension,
```

Comment:
What is meant by "default" ?

Reply:
It is defined in `cw721` package: `pub type DefaultOptionalNftExtension = Option<NftExtension>;`

So it is optional onchain metadata.